### DISCUSSION TITLE
Make filtering easier

### DISCUSSION THREAD

**Discussion by ImaCrea** [directus/directus#17872] [Feature Requests]
### Summary

# problem 

The filter module is awesome. Though, I believe it could be simpler for app user who haven't modeled the base structure.

Let's say I have a list of contacts and inside there's a O2M field linked to a collection of Organization. Pretty simple.

At the moment, when I'm on the Contacts collection and I want to see only people who are working at "ACME" corp. I can't search (because search doesn't work on relational field) so I have to filter. Alright.

But then I start to filter and what happens? Filtering on "Organization" is not enough. I have to go deeper and find the field "Org. name" into the object "Organization" to make my filter... Most users I tested stops at the first layer "Organization" which is totally understandable if they don't have much knowledge with databases.

# idea

A solution could be, as a system admin, to let me define the input "Org. name" in collection "Organization" as the *"default input to filter"*.

### Motivation

Ease the UX for average user with no particular db knowledge.

### Detailed Design

I can think about something more elaborate but I'd like to see some response on this topic first :)

### Requirements List

I can think about something more elaborate but I'd like to see some response on this topic first :)

### Drawbacks

I can think about something more elaborate but I'd like to see some response on this topic first :)

### Alternatives

I can think about something more elaborate but I'd like to see some response on this topic first :)

### Adoption Strategy

I can think about something more elaborate but I'd like to see some response on this topic first :)

**Comment by u12206050** [comment]
I agree, and in addition be able to hide which fields are actually selectable, since some just make more sense than others to have in the filter at all.

**Reply by carwei** [reply]
Great addition!

**Reply by u12206050** [reply]
Another addition within the same scope I believe is setting explicitly which columns/fields should be searched when the using the search bar. This way search could be improved, and even include searching translations for example on articles where by specifying that the relationship should be searched.

**Comment by carwei** [comment]
I like this proposal a lot. Been watching real users use the Admin app, and it's not clear to them that they need to drill down into the filter menu actually to use the filters. They type something in and complain that it doesn't work. They don't understand that Directus expected an ID at the first level. When they do drill down, they are confused with all the options that appear on the next level. The current experience is not smooth enough for a non-forgiving, non-tech user.

**Reply by ImaCrea** [reply]
exacty

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by rijkvanzanten** [comment]
@ImaCrea Looks like there's plenty of engagement for this idea 🙂  Mind filling out the blanks in the RFC so we can review properly? Thanks!
